ENH: add provider column to us_covid table

sglyon opened the pull request.

sglyon committed:
ENH: add provider column to us_covid table

cc7768 commented:
Quick thoughts based on conversation we had:

1. I think it will make more sense to separate the `valorum` provider into `state` and `county` providers where `state` means that the data was obtained from a state dashboard and `county` means that the data was obtained from a county dashboard. The default value for `provider` can be `state`
2. The current insert query which only inserts if no other provider has the data, is limiting since it will mean that the data will be incomplete for providers which have data for geographies that that we scrape on our own. Per our discussion, it may make sense just to "embrace" the data duplication and store each provider in their own table in the `data` schema and then have a trigger that automatically attempts to insert the data into a combined `data.us_covid` table which incorporates data from all possible providers. This means that we should add a new table for the `state` and `county` providers.
3. We will need to add the provider to the insert query for any data we scrape from a county provider.
4. We didn't explicitly talk about this but this may also be a good time to simplify the covid variables that we collect.

sglyon committed:
ENH: insert providers into separate table

Also copy into us_covid via trigger function

sglyon committed:
ENH: split valorum provider into state and county

Default is `state`.

Update county dashbaord scrapers to have provider be `county`

sglyon committed:
BUG: fix bug in NYT insert query

sglyon commented:
Thanks for summarizing.

I finished implementing 1-3 

I didn't do 4 yet -- I voted it belongs in a separate PR

cc7768 reviewed (approved):
All changes look good. I think you said you were going to add one more commit -- I'll take a look once it's there.

sglyon committed:
TEST: test provider trigger function logic
